[bug] Android dev fail in Windows

### Describe the bug

`tauri android dev` throw an error:

```
error Command failed with exit code 4294967295.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
       Error Failed to assemble APK: command ["D:\\workspace\\rust\\tauri\\examples\\api\\src-tauri/gen/android\\api\\gradlew.bat", "--stacktrace", "--project-dir", "D:\\workspace\\rust\\tauri\\examples\\api\\src-tauri/gen/android\\api"] exited with code 1
error Command failed with exit code 1.
```

Gradle output:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:rustBuildX86_64Debug'.
> A problem occurred starting process 'command 'yarn''

...

Caused by: java.io.IOException: Cannot run program "yarn" (in directory "D:\workspace\rust\tauri\examples\api\src-tauri"): CreateProcess error=2, 系统找不到指定的文件 (translated: 'The system cannot find the file specified')。
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
        ... 7 more
Caused by: java.io.IOException: CreateProcess error=2, 系统找不到指定的文件 (translated: 'The system cannot find the file specified')。
```

It means that the gradle process can not find the `npm` or `yarn` command, but why?

This did not happen in `@tauri-apps/cli@2.0.0-alpha.2`


### Reproduction

1. Development in Windows with Node installed.
2. Clone tauri repo and checkout `next` branch.
3. Build CLI + API.
4. Run `yarn tauri android init` + `yarn tauri android dev` at `examples/api`.

### Expected behavior

Android apk should be compiled successfully.

### Platform and versions

```text
Environment
  › OS: Windows 10.0.22624 X64
  › Webview2: 111.0.1661.44
  › MSVC: 
      - Visual Studio Community 2022
  › Node.js: 16.19.1
  › npm: 8.19.3
  › pnpm: 7.29.0
  › yarn: 1.22.19
  › rustup: 1.25.2
  › rustc: 1.67.1
  › cargo: 1.67.1
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 2.0.0-alpha.3
  › @tauri-apps/api [NPM]: 2.0.0-alpha.1
  › tauri [RUST]: 2.0.0-alpha.4,
  › tauri-build [RUST]: 2.0.0-alpha.2,
  › tao [RUST]: 0.18.0,
  › wry [RUST]: 0.27.0,

App
  › build-type: bundle
  › CSP: style-src 'unsafe-inline' 'self' https://fonts.googleapis.com; default-src 'self' customprotocol: asset:; font-src https://fonts.gstatic.com; img-src 'self' asset: https://asset.localhost blob: data:
  › distDir: ../dist
  › devPath: http://localhost:5173/
  › framework: Svelte
  › bundler: Vite

App directory structure
  ├─ dist
  ├─ isolation-dist
  ├─ node_modules
  ├─ public
  ├─ src
  └─ src-tauri
```


### Stack trace

_No response_

### Additional context

_No response_

worked fine for me, can you try recreating the android project, clearing the target folder and potentially any old apks on the emulator with the same bundler identifier? 

I did encounter another bug but that bug will crash the CLI and it is with `pnpm`, so I am gonna assume it is not related to this, but here is the error in case you run into it:
```

> tauri-app@0.0.0 tauri D:\dev\tauri-app
> tauri "--" "android" "android-studio-script" "--target" "x86_64"

error: unexpected argument 'android' found

  note: subcommand 'android' exists; to use it, remove the '--' before it

Usage: pnpm run tauri [OPTIONS] <COMMAND>

For more information, try '--help'.
?ELIFECYCLE? Command failed with exit code 2.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:rustBuildX86_64Debug'.
> Process 'command 'pnpm'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
 ELIFECYCLE  Command failed with exit code 4294967295.
       Error Failed to assemble APK: command ["D:\\dev\\tauri-app\\src-tauri/gen/android\\tauri_app\\gradlew.bat", "--project-dir", "D:\\dev\\tauri-app\\src-tauri/gen/android\\tauri_app"] exited with code 1
```

--- a/.changes/napi-rs.md
+++ b/.changes/napi-rs.md
@@ -1,0 +1,5 @@
+---
+'cli.js': 'patch'
+---
+
+Update `napi-rs` dependencies to latest to fix CLI hanging up forever.

--- a/tooling/cli/node/Cargo.toml
+++ b/tooling/cli/node/Cargo.toml
@@ -11,8 +11,8 @@ openssl-vendored = ["tauri-cli/openssl-vendored"]
 
 [dependencies]
 # Default enable napi4 feature, see https://nodejs.org/api/n-api.html#node-api-version-matrix
-napi = { version = "2.10", default-features = false, features = ["napi4"] }
-napi-derive = "2.10"
+napi = { version = "2.12", default-features = false, features = ["napi4"] }
+napi-derive = "2.12"
 tauri-cli = { path = ".." }
 log = "0.4.17"
 

--- a/tooling/cli/node/package.json
+++ b/tooling/cli/node/package.json
@@ -37,7 +37,7 @@
     }
   },
   "devDependencies": {
-    "@napi-rs/cli": "2.14.4",
+    "@napi-rs/cli": "2.15.1",
     "cross-env": "7.0.3",
     "cross-spawn": "7.0.3",
     "fs-extra": "11.1.0",

